### PR TITLE
[16.0][MIG] mgmtsystem_claim

### PR DESCRIPTION
**Pull request by mymage**
depends https://github.com/OCA/crm/pull/468

### CODE REVIEW

File: mgmtsystem_claim/views/res_partner_views.xml
Comment (ivantodorovich) [inline mgmtsystem_claim/views/res_partner_views.xml]: Don't add the `base.group_partner_manager` here. It'll change the previous behaviour.
I don't know why it was added to the view `groups_id` before, but only users belonging to `sales_team.group_sale_salesman` were able to see this button (as it was the only group defined in the element's `groups` attribute). So it shuold remain like this, IMO

Comment (mymage) [reply mgmtsystem_claim/views/res_partner_views.xml]: Following  the migration instruction, I deleted
```
<field
            name="groups_id"
            eval="[(4, ref('sales_team.group_sale_salesman')), (4, ref('base.group_partner_manager'))]"
        />
```
adding the "base.group_partner_manager" to the already present groups parameter.
I'm not confident with this change, so if you are sure, I trust you and delete it.

Comment (ivantodorovich) [reply mgmtsystem_claim/views/res_partner_views.xml]: Yes, please.
The explaination is this:


**Before:**

- The `ir.ui.view` was applied for both groups
- The button was shown only for `sales_team.group_sale_salesman`
- _Effectively, only users belogning to `sales_team.group_sale_salesman` saw the button_

**With your change:**

- The `ir.ui.view` is applied for all groups (normal, as Odoo deprecated the use of `groups_id`)
- The button is shown for both `sales_team.group_sale_salesman` and `base.group_partner_manager`


---

As you can see, the behaviour has changed.
To keep the previous behaviour we simply need to remove the `group_id` from the `ir.ui.view`, and keep the previous value for `groups` on the button element

Comment (mymage) [reply mgmtsystem_claim/views/res_partner_views.xml]: Thanks for explanation. ASAP I will fix tests using "remote" pr on crm.

**Review by primes2h — CHANGES_REQUESTED** [review]
This change must be moved in the migration commit.

File: mgmtsystem_claim/views/res_partner_views.xml
Comment (primes2h) [inline mgmtsystem_claim/views/res_partner_views.xml]: This change must be moved in the migration commit

File: test-requirements.txt
Comment (primes2h) [inline test-requirements.txt]:
```suggestion
odoo-addon-crm_claim @ git+https://github.com/OCA/crm/pull/468/head#subdirectory=setup/crm_claim
```

**Review by primes2h — CHANGES_REQUESTED** [review]

File: mgmtsystem_claim/views/res_partner_views.xml
Comment (mymage) [reply mgmtsystem_claim/views/res_partner_views.xml]: Yes I know, but I'm lazy ;-)

Comment (ivantodorovich) [reply mgmtsystem_claim/views/res_partner_views.xml]: Still, he makes a good point and it should be fixed 😅 
It's currently in the `[DONT MERGE] test-requirements.txt` commit, which will be removed after the dependency is merged. We don't want to loose these changes by mistake so better to put them in the right commit 😄

Comment (mymage) [reply mgmtsystem_claim/views/res_partner_views.xml]: ok guys, done

File: test-requirements.txt
Comment (mymage) [reply test-requirements.txt]: the solution was:
odoo-addon-crm_claim @ git+https://github.com/OCA/crm.git@refs/pull/468/head#subdirectory=setup/crm_claim

**Comment by FernandoRomera** [comment]
@mymage 
Done https://github.com/OCA/crm/pull/468

**Review by dreispt — APPROVED** [review]

**Comment by etobella** [comment]
You removed groups_id on a view. Might be necessary to add a migration script, as it will not be changed when migrating on old databases, making two different behaviours

**Comment by dreispt** [comment]
@mymage Maybe you need to add the  `<t groups="...">` element?

**Comment by mymage** [comment]
@etobella you have too much faith in my abilities :-) but if it the case I'll learn how to do it
@dreispt I hope I have interpreted your suggestion in the right way and that this avoids the route suggested by @etobella

**Comment by ivantodorovich** [comment]
> You removed groups_id on a view. Might be necessary to add a migration script, as it will not be changed when migrating on old databases, making two different behaviours

AFAICS there's no migration script needed.

Before, the view had the field `groups_id` set, it's true.
This is a field Odoo deprecated. We're no longer able to set groups on ir.ui.view records
We must set the groups on the view's arch elements, instead.

However, the button element already had a group defined: `groups="sales_team.group_sale_salesman"`
So in fact the groups in the view record was just redundant

**Review by ivantodorovich — APPROVED** [review]

**Comment by primes2h** [comment]
@mymage 
Please merge/squash commits generated by bots. See https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate

**Pull request merged**
